packages/bsync/src/purchases/purchasesClient.ts

```diff
@@ -0,0 +1,272 @@
+      },
+      ios: {
+        'core:monthly': 'bluesky_plus_core_v1_monthly',
+        'core:annual': 'bluesky_plus_core_v1_annual',
+      },
+      android: {
+        'core:monthly': 'bluesky_plus_core_v1:monthly',
+        'core:annual': 'bluesky_plus_core_v1:annual',
+      },
+    }
+
+    this.GROUPS = {
+      core: {
+        web: [
+          { id: 'core:monthly', product: this.PRODUCTS.web['core:monthly'] },
+          { id: 'core:annual', product: this.PRODUCTS.web['core:annual'] },
+        ],
+        ios: [
+          { id: 'core:monthly', product: this.PRODUCTS.ios['core:monthly'] },
+          { id: 'core:annual', product: this.PRODUCTS.ios['core:annual'] },
+        ],
+        android: [
+          {
+            id: 'core:monthly',
+            product: this.PRODUCTS.android['core:monthly'],
+          },
+          { id: 'core:annual', product: this.PRODUCTS.android['core:annual'] },
+        ],
+      },
+    }
+  }
+
+  isRcWebhookAuthorizationValid(authorization: string | undefined): boolean {
+    return this.revenueCatClient.isWebhookAuthorizationValid(authorization)
+  }
+
+  getSubscriptionGroup(group: string, platform: string) {
+    assertSubscriptionGroup(group)
+    assertPlatform(platform)
+    return this.GROUPS[group][platform]
+  }
+
+  async getEntitlements(did: string): Promise<string[]> {
+    const { subscriber } = await this.revenueCatClient.getSubscriber(did)
+
+    const now = Date.now()
+    return Object.entries(subscriber.entitlements)
+      .filter(
+        ([_, entitlement]) =>
+          now < new Date(entitlement.expires_date).valueOf(),
+      )
+      .map(([entitlementIdentifier]) => entitlementIdentifier)
+  }
+
+  async getSubscriptionsAndEmail(
+    did: string,
+  ): Promise<{ subscriptions: Subscription[]; email?: string }> {
+    const { subscriber } = await this.revenueCatClient.getSubscriber(did)
+
+    const email = subscriber.subscriber_attributes.email?.value
+
+    const subscriptions = Object.entries(subscriber.subscriptions)
+
+    return {
+      email: email,
+      subscriptions: subscriptions
+        .map(this.createSubscriptionObject)
+        .filter(Boolean) as Subscription[],
+    }
+  }
+
+  private createSubscriptionObject = ([productId, s]: [
+    string,
+    RcSubscription,
+  ]): Subscription | undefined => {
+    const platform = mapStoreToPlatform(s.store)
+    if (!platform) {
+      return undefined
+    }
+
+    const fullProductId =
+      platform === 'android'
+        ? `${productId}:${s.product_plan_identifier}`
+        : productId
+
+    const group = this.mapStoreIdentifierToSubscriptionGroup(fullProductId)
+    if (!group) {
+      return undefined
+    }
+
+    const now = new Date()
+    const expiresAt = new Date(s.expires_date)
+
+    let status = 'unknown'
+    if (s.auto_resume_date) {
+      if (now > expiresAt) {
+        status = 'paused'
+      }
+    } else if (now > expiresAt) {
+      status = 'expired'
```

Comment:
Does this seem accurate?
```suggestion
    } else if (now >= expiresAt) {
      status = 'expired'
```